### PR TITLE
Reorganize Host Settings

### PR DESCRIPTION
**Pull request by jeffin143**
Resolve https://github.com/opencollective/opencollective/issues/3012

# Description

This work is mostly about moving settings that are already existing.

**User or Organizations (that ARE NOT yet hosts) should now see:**

```
...
Webhooks
Advanced

Fiscal Hosting
 - Activate (moved from advanced) - Done
```

**User or Organizations (that ARE hosts) should see:**

```
...
Webhooks
Advanced

Fiscal Hosting
  - Enable Applications (moved from advanced) - Done
  - Host fee (moved from advanced) - Done
  - Terms of Fiscal Sponsorship (moved from info, was named Terms of service) - Done
  - Deactivate (moved from advanced) - Done
Host Plan
  - Same as existing
Expenses & Payouts
  - Expense Policy (same as existing)
Invoices & Receipts
  - Same as Existing
Receiving Money
 - Connect Stripe (moved from Connected Accounts) - Done
 - Bank Transfers (moved from Payment Methods) - Done 
   - BONUS: Users should be able to delete Bank Transfers settings 
Sending Money
 - PayPal: for now, just explain that PayPal is activated by default - Done 
 - Transferwise (moved from Connected Accounts) - Done 
```

### CODE REVIEW

**Comment by jeffin143** [comment]
How to test transferwise option

**Comment by znarf** [comment]
> How to test transferwise option

According to:

```
  if (props.collective.isHost && hasFeature(props.collective, FEATURES.TRANSFERWISE)) {
    services.push('transferwise');
  }
```

The Host should have the `settings.features.transferwise = true`, otherwise, you can also just comment the condition in developer environment.

**Comment by jeffin143** [comment]
@znarf , This is almost done, except 
```
Fiscal Hosting
 - Activate (moved from advanced)
``` 

Could you point me to this option, I tried making a collective and din't find this button , may be I missed something

**Comment by znarf** [comment]
@jeffin143 This is only for User or Organization, try creating on Organization, you will find it in "Advanced" section.

Please double check that the changes are only for Hosts (**User** and **Organization**) and are not affecting **Collective** settings.

When you're done, don't forget to run `npm run langs:update` and commit the result.

**Comment by jeffin143** [comment]
> @jeffin143 This is only for User or Organization, try creating on Organization, you will find it in "Advanced" section.
> 
> Please double check that the changes are only for Hosts (**User** and **Organization**) and are not affecting **Collective** settings.
> 

@znarf , This is Done, I did change the activate button but wasn't able to test it, and hence I queried it, But I did test now and it works fine. 

So this is ready to be reviewed now :)

**Comment by znarf** [comment]
Great!

Spotted a few more things that are not exactly matching the specs.

- [ ] "Host Plan" should be below "Fiscal Hosting"
- [ ] The entry menu for "Expenses Policy" should be "Expenses & Payouts"
  - [ ] The page for "Expenses & Payouts"" should have a title "Expenses & Payouts", "Expenses Policy" should still be there as a label for the Textarea

Generic layout enhancements:
- [ ] The section title (Fiscal Hosting, Host Plan, Expenses & Payouts) should always appear at the same place and should be in the same font.

Small enhancement for the "Fiscal Hosting" section:
- [ ] Deactivate section on the bottom.
- [ ] "Save" button should not be visible if "Fiscal Hosting" is not yet activated

The layout should be the following:

```
=Fiscal Hosting=

Applications [       ]
Host Fee [       ]
Terms of fiscal sponsorship [       ]

Save

-----------------------------------------

==Deactivating as host==
(text)
[Deactivate as Host]
```

File: components/edit-collective/MenuEditCollective.js [hunk 1]
```diff
@@ -31,6 +31,9 @@ export const EDIT_COLLECTIVE_SECTIONS = {
   // Host Specific
   HOST_SETTINGS: 'hostSettings',
   INVOICES: 'invoices',
+  RECEIVING_MONEY: 'receiving-money',
+  SENDING_MONEY: 'sending-money',
+  FISCALHOSTING: 'fiscalhosting',
```
Comment (znarf) [inline components/edit-collective/MenuEditCollective.js]:
```suggestion
  FISCAL_HOSTING: 'fiscal-hosting',
```

Comment (jeffin143) [reply components/edit-collective/MenuEditCollective.js]: Done

**Comment by jeffin143** [comment]
@znarf , Can you check why the createvent test is failing, I have no issue in my local in order to create event

**Comment by znarf** [comment]
@jeffin143 have you tried to run the E2E locally ? https://github.com/opencollective/opencollective-frontend/blob/master/docs/e2e.md

**Comment by jeffin143** [comment]
Yes I did run it with cypress:open too and passed successfully

**Comment by znarf** [comment]
@jeffin143 the "Save" button is missing when editing Tickets (at least)

Capture d’écran 2020-04-02 à 10 50 21

**Comment by jeffin143** [comment]
Really sorry @znarf , My bad I have been testing at the wrong branch all this time. :'(

**Comment by jeffin143** [comment]
Let's wait for 7 mins and see this if it passes

**Comment by znarf** [comment]
@jeffin143 Almost there! From the previous checklist, please review:
- The page for "Expenses & Payouts"" should have a title "Expenses & Payouts", "Expenses Policy" should still be there as a label for the Textarea

Also:
- The size of the font for "Invoices & Receipts" is different from the others (1.7 vs 1.8)

**Comment by jeffin143** [comment]
> * The page for "Expenses & Payouts"" should have a title "Expenses & Payouts", "Expenses Policy" should still be there as a label for the Textarea

Sorry I miss understood that comment , I thought accessibility should be expenses policy I have made apt changes 

> 
> * [ ]  The size of the font for "Invoices & Receipts" is different from the others (1.7 vs 1.8)

I have made it consistent 1.7 for all of them (Host Plan, Fiscal Hosting, Sending Money, Receiving Money , Invoices and Receipts )

File: components/edit-collective/EditHostSettings.js [hunk 2]
```diff
@@ -115,9 +116,9 @@ const EditHostSettings = props => {
 
   return (
     <div className="ExportData">
-      <h3>
+      <H3>
         <FormattedMessage id="Host.Plan" defaultMessage="Host Plan" />
-      </h3>
+      </H3>
```
Comment (jeffin143) [inline components/edit-collective/EditHostSettings.js:121]: Host plan was H3 and hence I have changed it all to H3

File: components/edit-collective/EditPaymentMethods.js [hunk 3]
```diff
@@ -225,11 +229,11 @@ class EditPaymentMethods extends React.Component {
...
           </MessageBox>
         )}
-        {!showManualPaymentMethodForm && (
+        {this.props.sendingSection && !showManualPaymentMethodForm && (
           <Flex className="paymentMethods" flexDirection="column" my={2}>
-            <H2>
+            <H3>
               <FormattedMessage id="paymentMethods.send.title" defaultMessage="Sending money" />
-            </H2>
+            </H3>
```
Comment (jeffin143) [inline components/edit-collective/EditPaymentMethods.js:236]: H3

File: components/edit-collective/EditPaymentMethods.js [hunk 4]
```diff
@@ -274,15 +278,52 @@ class EditPaymentMethods extends React.Component {
             </Span>
           </Flex>
         )}
-        {showEditManualPaymentMethod && (
+        {this.props.receivingSection && showEditManualPaymentMethod && (
           <React.Fragment>
-            <H2>
+            <H3>
               <FormattedMessage id="paymentMethods.receive.title" defaultMessage="Receiving money" />
+            </H3>
```
Comment (jeffin143) [inline components/edit-collective/EditPaymentMethods.js:285]: H3

File: components/edit-collective/EditCollectiveForm.js [hunk 5]
```diff
@@ -794,73 +835,128 @@ class EditCollectiveForm extends React.Component {
...
-                              context={this.state.collective}
-                              onChange={value => this.handleChange(field.name, value)}
-                            />
-                          ),
-                      )}
+            {this.state.section === EDIT_COLLECTIVE_SECTIONS.FISCAL_HOSTING && (
+              <H3>
+                {' '}
+                <FormattedMessage id="Fiscalhost" defaultMessage={'Fiscal Host'} />{' '}
+              </H3>
```
Comment (jeffin143) [inline components/edit-collective/EditCollectiveForm.js:842]: And H3

**Comment by znarf** [comment]
>> The page for "Expenses & Payouts"" should have a title "Expenses & Payouts", "Expenses Policy" should still be there as a label for the Textarea

> Sorry I miss understood that comment , I thought accessibility should be expenses policy I have made apt changes

There should still be an "Expenses & Payouts" title on top of "Expenses Policy".

**Comment by jeffin143** [comment]
Screenshot 2020-04-02 at 6 48 42 PM

Done

**Comment by znarf** [comment]
Sorry, "Sending Money" is still `1.8` while other titles are `1.7`. Then maybe "PayPal" needs to be like the other second level titles too.

Also, can you make sure to rebase with master?

Thank you so much 🙏

**Comment by znarf** [comment]
Looking good! Will do a final review tomorrow, just to be sure there is no missed blocker.

**Comment by znarf** [comment]
@jeffin143 I merged but I just found something that is not good for the future.

The relation between `EditPaymentMethods` and `EditReceivingSendingMoney` is highly confusing.

- `EditPaymentMethods` should only be about Credit Cards
- "Bank Transfers" should be moved out of it and extracted in a Component.

**Pull request merged**